libknet/threads_tx.c

```diff
@@ -349,7 +349,7 @@ static int _parse_recv_from_sock(knet_handle_t knet_h, size_t inlen, int8_t chan
 			       knet_h->send_to_links_buf_compress, (ssize_t *)&cmp_outlen);
 		if (err < 0) {
 			log_warn(knet_h, KNET_SUB_COMPRESS, "Compression failed (%d): %s", err, strerror(errno));
-		} else {
+		} else if (cmp_outlen < inlen) {
```

Comment:
I don't like this bit. We need to account for all calls into compress whether data was compressed or not, that's all part of the latency calculation.

We already account for uncompressed packets in the stats but not 'failed' compressions - it's an unlikely event but worth keeping stats on I suppose .. but yes, not in the stable branch,.

Reply:
If we are going to change the stats, we could have 2 extra counters:

failed_to_compress (errors from the compression libraries)
unable_to_compress (compress is bigger than original packet)

both can be useful I think,but since it's an ABI change it cannot go back into stable branch.

@ppiao can you split this PR into two? one for the extra counters and one to fix the uncompressed_packet conditional fix?